RELEASE.md

```diff
@@ -8,7 +8,17 @@ We do not guarantee backports from the `main` branch to older release branches.
 
 This differs from the previous release schedule, which was driven by OpenShift releases.
 
-# How to cut a new release
+| Release | Kubernetes Version Tested | Release shepherd                        |
+|---------|---------------------------|-----------------------------------------|
+| v0.18.0 | 1.39, 1.38, 1.37          | **searching for volunteer**             |
```

Comment:
Considering that the current kube-prometheus covers Kube v1.x, v1.x+1 and v1.x+2, my expectation was that the next minor kube-prometheus would support v1.x+1, v1.x+2 and v1.x+3. Here we seem to skip one Kube release out of 2.

Reply:
I was looking at existing releases and respective go.mod file, also e2e test kind image versions. IIUC we are releasing currently skipping a version but we are testing a release prior in it. For example current main points to 1.33 but in e2e tests we have tests till 1.31. So I was adding a similar pattern. 

Let me know how you would want to change it :)

Reply:
I have no strong opinion but I suspect that people will ask for a release branch on every new Kubernetes version. Deferring to @philipgough for the decision :)

Reply:
I am ok with releasing new version for every k8s version going forward. Entries were based on previous releases :)

Reply:
I changed to version go.mod is using as we already have compatibility matrix which says about versions tested